src/utils/validateAvaxBurnedAmountEtna.ts

```diff
@@ -0,0 +1,58 @@
+import type { Context } from '../vms/context/model';
+import {
+  isAddPermissionlessDelegatorTx,
+  isAddPermissionlessValidatorTx,
+  isAddSubnetValidatorTx,
+  isCreateChainTx,
+  isCreateSubnetTx,
+  isPvmBaseTx,
+  isExportTx as isPvmExportTx,
+  isImportTx as isPvmImportTx,
+  isRemoveSubnetValidatorTx,
+  isTransferSubnetOwnershipTx,
+} from '../serializable/pvm';
+import type { UnsignedTx } from '../vms/common';
+import type { FeeState } from '../vms/pvm';
+import { calculateFee } from '../vms/pvm/txs/fee/calculator';
+
+export const validateAvaxBurnedAmountEtna = ({
+  unsignedTx,
+  context,
+  burnedAmount,
+  feeState,
+}: {
+  unsignedTx: UnsignedTx;
+  context: Context;
+  burnedAmount: bigint;
+  feeState: FeeState;
+}): { isValid: boolean; txFee: bigint } => {
+  const tx = unsignedTx.getTx();
+
+  const expectedFee = calculateFee(
+    unsignedTx.getTx(),
+    context.platformFeeConfig.weights,
+    feeState.price < context.platformFeeConfig.minPrice
+      ? context.platformFeeConfig.minPrice
+      : feeState.price,
+  );
+
+  if (
+    isPvmBaseTx(tx) ||
+    isPvmExportTx(tx) ||
+    isPvmImportTx(tx) ||
+    isAddPermissionlessValidatorTx(tx) ||
+    isAddPermissionlessDelegatorTx(tx) ||
+    isAddSubnetValidatorTx(tx) ||
+    isCreateChainTx(tx) ||
+    isCreateSubnetTx(tx) ||
+    isRemoveSubnetValidatorTx(tx) ||
+    isTransferSubnetOwnershipTx(tx)
+  ) {
+    return {
+      isValid: burnedAmount >= expectedFee,
```

Comment:
This is not very comforting. The purpose of this utility is to make sure we are not burning too much avax, but this provides no such comfort. I would look at the EVM burn checker for a good pattern on how to ensure that the burn amount is in a reasonable window

Reply:
adjusted the logic to set a window between expectedFee <= burnedAmount <= expectedFee * tolerance